### PR TITLE
Pr oe aineiston siirto in timeen 282

### PR DESCRIPTION
**Pull request by tahtee**

### CODE REVIEW

**Review by mkulma — APPROVED** [review]

File: service/src/main/kotlin/fi/ouka/evakaoulu/invoice/service/EVakaOuluInvoiceClient.kt [hunk 1]
```diff
@@ -4,14 +4,44 @@
...
+        withSSN.forEach {
+            proEinvoices += invoiceGenerator.generateInvoice(it)
+            successList.add(it)
+        }
+
+        try {
+            manuallySentList.addAll(withoutSSN)
+            invoiceSender.send(proEinvoices)
+        } catch (e: SftpException){
+            failedList.addAll(successList)
```
Comment (mrosin) [inline service/src/main/kotlin/fi/ouka/evakaoulu/invoice/service/EVakaOuluInvoiceClient.kt:33]: Suggest to add logging, at the very least here in error clause

File: service/src/test/kotlin/fi/ouka/evakaoulu/invoice/service/InvoiceSenderTest.kt [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+package fi.ouka.evakaoulu.invoice.service
+
+import fi.ouka.evakaoulu.EvakaOuluProperties
+import fi.ouka.evakaoulu.IntimeProperties
+import org.junit.jupiter.api.Test
+import org.mockito.kotlin.*
+
+internal class InvoiceSenderTest {
+
```
Comment (mrosin) [inline service/src/test/kotlin/fi/ouka/evakaoulu/invoice/service/InvoiceSenderTest.kt:13]: Suggestion: When working with clients it´s good to write/have a test for the most common exception -> connection timeout/ not available

**Review by mrosin — CHANGES_REQUESTED** [review]

File: service/src/test/kotlin/fi/ouka/evakaoulu/invoice/service/InvoiceSenderTest.kt (hunk 2 above)
Comment (tahtee) [reply service/src/test/kotlin/fi/ouka/evakaoulu/invoice/service/InvoiceSenderTest.kt:13]: This is already tested with this 

```
@Test
    fun `should return unsuccessfully sent invoices in failed list`() 
```

File: service/src/main/kotlin/fi/ouka/evakaoulu/invoice/service/EVakaOuluInvoiceClient.kt (hunk 1 above)
Comment (tahtee) [reply service/src/main/kotlin/fi/ouka/evakaoulu/invoice/service/EVakaOuluInvoiceClient.kt:36]: Logging added.

**Review by vesahonkasaari — APPROVED** [review]

**Pull request merged**